### ISSUE TITLE
Key/cert rotation to facilitate pinning

### ISSUE THREAD

**Issue by timbray** [keybase/keybase-issues#964]
It’s been suggested, for the work I’m doing on OpenKeychain as a Keybase client, that I pin the https connection to Keybase’s cert/key, for example see http://thoughtcrime.org/blog/authenticity-is-broken-in-ssl-but-your-app-ha/ and https://www.owasp.org/index.php/Certificate_and_Public_Key_Pinning

This would allow sidestepping a bunch of potential stinky-CA-biz vulnerabilities and seems like a good idea to me in general; after all, I’m already compiling the base URI into the app code.  But of course the problem is that this depends on working with Keybase’s key/cert rotation policy, if any, and in an open-source project where you can’t depend on someone being around to update the client.  (Google’s policy, IIRC, was that they rotated the certs regularly but stayed with the same key.)  My spidey-sense is also tingling, wondering if Keybase’s own infrastructure might be useful in making this easier and/or more secure.

If there’s something there already about keys/certs, I didn’t turn it up.

**Comment by dschuermann** [comment]
@maxtaco I am currently pinning keybase.io in Openkeychain and am confused. The API docs only talk about https://keybase.io not https://api.keybase.io . So the certificate from https://github.com/keybase/node-client/blob/master/src/ca.iced is not valid. I am now pinning the CA of keybase.io for now...
Would be nice if you resolve this inconsistency.

**Comment by maxtaco** [comment]
Indeed, the CA from that link is only valid for `api.keybase.io`, which is what our Node client uses (and what our Go code will also use).  For those using shell scripts, or Web clients, etc, where it's no convenient to ship with a CA cert, we suggest keybase.io, which is signed via RapidSSL.  If you're using that URL, it ought to be pinned as you say.

**Comment by dschuermann** [comment]
@maxtaco can we switch to api.keybase.io for OpenKeychain? Is it the same API just a different domain?

**Comment by maxtaco** [comment]
It's the same API served by the same server, with just a different domain.  I think switching to api.keybase.io is a great idea.  I'll put it on our list to document this issue better on the web side.

**Comment by dschuermann** [comment]
@maxtaco thx, done in https://github.com/open-keychain/open-keychain/commit/3a9709a35d3cf5cd2e93e49b75745226fd59ae2d
